### ISSUE TITLE
libtorrent 32-bit Windows build segfaults on partial `resume_data`

### ISSUE THREAD

**Issue by kotenok2000** [Tribler/tribler#8353]
**Traceback and/or steps to reproduce**
Tribler 8.0.7 crashes when i attempt to import a torrent with already existing downloaded files from an old version.
I pressed "Import torrent as a file". opened .torent file and set destination to F:\TriblerDownloads\TRON RUN-r [FitGirl Repack]\
```python
Insert your traceback here!
```
[CrashDumps.zip](https://github.com/user-attachments/files/18233150/CrashDumps.zip)

**Comment by qstokkink** [comment]
Thanks for the report! I haven't been able to reproduce this yet: if you have any more information on what triggers this crash, please let us know.

**Comment by qstokkink** [comment]
Alright, I tried just about every variation of importing old downloads. I corrupted the files (downloaded, conf, state, parts), added odd file names (spaces, special characters), etc.

I looked at your crashdumps again. Is it correct that you're using 32-bit Windows 10? I may be the case that one of our dependencies is secretly using 64-bit addressing and importing a torrent just happened to trigger it by accident.

**Comment by kotenok2000** [comment]
I am using 64 bit windows 11 but 32 bit tribler.

I have recreated the bug and it has created this file in dlcheckpoints.
Tribler crashes when it attempts to load it.

[54cd87fcfcdfcf51011164bd22d9e1e34ffc8b38.conf.txt](https://github.com/user-attachments/files/18507680/54cd87fcfcdfcf51011164bd22d9e1e34ffc8b38.conf.txt)

**Comment by qstokkink** [comment]
Thanks, I'll try to reproduce it using that conf file. So far:

 - Ubuntu: works.
 - Windows x64 using x64 Tribler: works.
 - Windows x64 using x32 Tribler: crashes.

**Comment by qstokkink** [comment]
My conclusion is that the `libtorrent` (both `1.2.19` and `2.0.9`) **32-bit** Windows build crashes (i.e., tries to access a NULL-pointer and SEGFAULTS) if, _and only if_, the `save_path` in the resume data is either unset or an empty string (**CC**: @arvidn).

I guess it's fine to just make sure the `b'save_path'` is always added (idea by @egbertbouman), even though it's not necessary for any other build.

**Comment by arvidn** [comment]
yeah, `save_path` is currently treated as a required argument, and there's an assert to make sure it's passed in. It's an oversight that this is not handled well when loading (possibly unreliable) resume data from disk.
I'll take a look.

**Comment by arvidn** [comment]
does this address the issue? It turns the invalid `add_torrent_params` into an exception when added to the session.
https://github.com/arvidn/libtorrent/pull/7825

**Comment by qstokkink** [comment]
@arvidn thanks! I'll check it out. I will have to make a setup to compile the 32-bit Windows Python wheel though, so it may take me a little while to verify. Please give me a few days.

**Comment by qstokkink** [comment]
@arvidn I built a wheel from your https://github.com/arvidn/libtorrent/pull/7826 PR and used it with our Tribler 32-bit build. The result is:

 - Tribler win32 now segfaults with and without a `save_path` in the `resume_data`?!
 - Adding a torrent (using `add torrent parameters`) without `resume_data` still works.

Looking at your PR, I can't quite make sense of that behaviour. Perhaps Tribler's resume data is very wrong? I'm using the following:

```python
import libtorrent
resume_data = libtorrent.bencode({
    b'file-format': b'libtorrent resume file',
    b'file-version': 1,
    b'info-hash': ...,  # Omitted for privacy
    b'save_path': b'F:/BiglyBT Downloads/'  # Also attempted with backslashes, and no trailing slash
})
```

This used to work before (in `2.0.9`) and no longer works:

```python
import libtorrent
resume_data = libtorrent.bencode({b"save_path": b"F:\\BiglyBT Downloads\\"})
```

**Comment by qstokkink** [comment]
We're passing fast resume data in the atp. According to https://github.com/arvidn/libtorrent/blob/RC_2_0/include/libtorrent/add_torrent_params.hpp#L389-L395, this is now deprecated. Maybe we just need to modernize our source code?

**Comment by arvidn** [comment]
> This used to work before (in 2.0.9) and no longer works:
> 
> import libtorrent
> resume_data = libtorrent.bencode({b"save_path": b"F:\\BiglyBT Downloads\\"})

What happens instead? It seems to work on linux (it's not expected to segfault)

**Comment by qstokkink** [comment]
This started appearing again (using libtorrent `2.0.11`).

**Comment by teddyrogers** [comment]
> My guess is that one of our alert handlers (related to peers???) is doing something libtorrent 32-bit "doesn't like". Of course, I'll stress again that libtorrent 64 bit has no issues.

@qstokkink why waste time and energy supporting 32bit in 2025?

**Comment by qstokkink** [comment]
I guess that's a fair point. @egbertbouman are you O.K. with resolving this issue by removing 32 bit support?

**Comment by egbertbouman** [comment]
@qstokkink Sure, let's remove it. We can always reconsider if users complain.

**Comment by qstokkink** [comment]
Alright, let's unalive the 32 bit version then. There is already an issue open for that (#8556), so I'll close this then (at least for now).
